### ISSUE TITLE
Readme - spruce me up!

### ISSUE THREAD

**Issue by mixmix** [ssbc/patchwork#611]
@mmckegg lets give it a good bit of care.

mention these concepts for new devs : 
- depject
- mutant
- mcss
- patchcore

**Comment by Marak** [comment]
I'm finding very little to no code comments in the software itself.

This combined with the unorthodox coding styles from using libraries like `mutant`, `depject`, and `depnest` are making it difficult to make code contributions.

I strongly recommend getting into the habit of documenting blocks of code to express their intention and actual effects. Writing good comments can be harder than writing code itself.

For example, I want to modify an API method for fetching posts by type and end up here:
 [https://github.com/ssbc/patchwork/blob/master/main-window.js#L17](https://github.com/ssbc/patchwork/blob/master/main-window.js#L17)

It feels like I have to read the entire documentation for `depnest` and`depject` to even have a basic understanding of whats is going on here. That's not good. Starting to add code comments in the project could greatly improve readability.

**Comment by mixmix** [comment]
Yeah we are interested in being more accessible. Do you think a clear intro
to depject + depnesr would be a good idea?

**Comment by Marak** [comment]
I feel like there is already decent documentation for these projects in their respective READMEs.

It would be good to have one or two simple sentences above each block of code in `patchwork` explaining what's about to happen.

I find a declarative style easier to read than an imperative style for code comments. Less is more. 

Here is [an example](https://github.com/Stackvana/microcule/blob/master/lib/plugins/spawn/index.js) of a rather complex file I created that has somewhat simple declarative code comments. It's not perfect, but it helps. I like to approach code comments as a story written to myself in the future where I forgot everything I wrote in the file. That way others can easily jump in later and figure out what's going on.
